app/cdap/components/AddTransformation/SelectColumnsList/index.tsx

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import {
+  Box,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+  Radio,
+  Typography,
+} from '@material-ui/core';
+import React, { useEffect, useRef, useState } from 'react';
+import { useStyles } from '../styles';
+import { SearchIcon } from '../iconStore';
+import { prepareDataQualtiy } from '../CircularProgressBar/utils';
+import DataQualityProgress from '../CircularProgressBar';
+import { NoDataSVG } from 'components/GridTable/iconStore';
+import T from 'i18n-react';
+
+export default function(props) {
+  const {
+    directiveFunctionSupportedDataType,
+    selectedColumnsCount,
+    columnData,
+    setSelectedColumns,
+    dataQuality,
+  } = props;
+  const [columns, setColumns] = useState(columnData);
+  const [dataQualityValue, setDataQualityValue] = useState(dataQuality);
+  const [selectedColumns, setSelectedColumn] = useState([]);
+  const [focused, setFocused] = useState(false);
+  const classes = useStyles();
+  const ref = useRef(null);
+
+  const no_match =
+    directiveFunctionSupportedDataType?.length > 0 &&
+    directiveFunctionSupportedDataType?.includes('all')
+      ? directiveFunctionSupportedDataType?.filter((el) => el == 'all')
+      : columns.filter((object1) => {
+          return directiveFunctionSupportedDataType?.some((object2) => {
+            return object2.includes(object1?.type[0]?.toLowerCase());
+          });
+        });
+
+  useEffect(() => {
+    const getPreparedDataQuality = prepareDataQualtiy(dataQuality, columnData);
```

Comment:
why do we need a variable declaration here, can we set the state directly? or does it make better sense like this? plz take a call and update me in comments

Reply:
add null check as well plz

Reply:
we need to calculate data quality so firstly we are calling function and storing it's value to variable and then setting state